### ISSUE TITLE
[Question] Hello friends, how can I use the parameter logits_processors to modify the current logit? Is there any interface?

### ISSUE THREAD

**Issue by Jungle0430** [mlc-ai/mlc-llm#2871]
## ❓ General Questions

Hello friends, how can I use the parameter logits_processors to modify the current logit? Is there any interface?

**Comment by MasterJH5574** [comment]
Hi @Jungle0430 thanks for the question! Would you mind describing a bit more on what you want to do? For example like how you want to update the logits.

**Comment by Jungle0430** [comment]
> Hi @Jungle0430 thanks for the question! Would you mind describing a bit more on what you want to do? For example like how you want to update the logits.

e.g. To embed a text watermark within the model's output, we configure the sampling process by setting the logits for tokens in the left half of the output table to -inf. Is there an available interface for logit_processor to facilitate these adjustments?

**Comment by Jungle0430** [comment]
> > Hi @Jungle0430 thanks for the question! Would you mind describing a bit more on what you want to do? For example like how you want to update the logits.
> 
> e.g. To embed a text watermark within the model's output, we configure the sampling process by setting the logits for tokens in the left half of the output table to -inf. Is there an available interface for logit_processor to facilitate these adjustments?

This left-right dividing line is determined according to the cumulative probability distribution. So I need to get all the logits that the model outputs before generating each token.
